src/Compiler/Microsoft.NET.Sdk.Razor.SourceGenerators/RazorSourceGenerator.cs

```diff
@@ -226,15 +226,28 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                 .Where(pair => pair.Right)
                 .Select((pair, _) => pair.Left);
 
+            var isAddComponentParameterAvailable = compilation.Select((compilation, _) =>
+            {
+                var renderTreeBuilder = compilation.GetTypeByMetadataName("Microsoft.AspNetCore.Components.Rendering.RenderTreeBuilder");
+                if (renderTreeBuilder is null)
+                {
+                    return false;
+                }
+
+                return renderTreeBuilder.GetMembers("AddComponentParameter").Any();
```

Comment:
Need to check accessibility here too. Don't think we need to get fancy and do checks like IVT. I would just limit to "is it public".